File: app/src/main/java/com/stevenfrew/beatprompter/SetListFilter.java
package com.stevenfrew.beatprompter;

import java.util.List;

class SetListFilter extends SongFilter
{
    SetListFilter(String name,List<SongFile> songs)
    {
        super(name,songs,false);
    }

    boolean containsSong(SongFile sf)
    {
        return mSongs.contains(sf);
    }

    public boolean equals(Object o) {
        return o instanceof SetListFilter && mName.equals(((SetListFilter) o).mName);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/SetListFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.song.SongInfo

open class SetListFilter internal constructor(
	name: String,
	songs: List<Pair<SongInfo, String?>>
) : SongFilter(name, songs, false) {
	fun containsSong(sf: SongInfo): Boolean = songs.any { it.first == sf }
	override fun equals(other: Any?): Boolean = other is SetListFilter && name == other.name
	override fun hashCode(): Int = javaClass.hashCode()
}